### ISSUE TITLE
The default value for an option can't have surrounding spaces

### ISSUE THREAD

**Issue by ludat** [docopt/docopt.rs#165]
In the online version I can add a default for an option like:
```
  --separator=<separator>  Separator string [default:  | ].
```

Gives me the default separator value of ` | ` (note the spaces). But the same in rust just gives me `|` (no spaces)

**Comment by BurntSushi** [comment]
Yes, the code [trims default values](https://github.com/docopt/docopt.rs/blob/master/src/parse.rs#L255).

I'm not sure what the correct behavior is. For example, I don't think `[default: a]` should result in a default value of ` a`. So it sounds like you're suggesting that only a single leading space character be trimmed? That doesn't feel so good. It might be better if docopt introduced the ability to quote default values..?

cc @keleshev

**Comment by ludat** [comment]
I agree that the best behavior would be some kind of quoting but IMO the behavior should be consistent until the RFC (or whatever way of changing the standard) gets approved and applied to each implementation.

**Comment by BurntSushi** [comment]
@ludat I don't think this particular behavior is actually specified. (This implementation of docopt does do things beyond the specification, such as supporting greedy matching and repeatable flags defined only in the "options" section.)

**Comment by ludat** [comment]
If I implement the quoting would you merge it? `[Default: " !"]`. Maybe tonight I'll give it a go.

**Comment by BurntSushi** [comment]
@ludat I suspect I would! Permitting an escaped quote, e.g., `\"`, would probably be good. Everything else can just be treated as-is to keep it simple.

If you need any help, I should be on IRC tonight (6pm-9pm EST). I hang around most of the `#rust` channels under the nick `burntsushi`.
